daemon/execdriver/driver.go

```diff
@@ -24,10 +24,21 @@ var (
 	ErrDriverNotFound          = errors.New("The requested docker init has not been found")
 )
 
-// StartCallback defines a callback function.
-// It's used by 'Run' and 'Exec', does some work in parent process
-// after child process is started.
-type StartCallback func(*ProcessConfig, int)
+// DriverCallback defines a callback function which is used in "Run" and "Exec".
+// This allows work to be done in the parent process when the child is passing
+// through PreStart, Start and PostStop events
+type DriverCallback func(*ProcessConfig, int) error
```

Comment:
Sure, I'm fine with that. I think I will just document in the updated comment if that's ok with you as this code has actually been here a long time (but only used internally as the start callback for pid info, which is what the `int` is)